grype/search/distro.go

```diff
@@ -53,10 +62,50 @@ func (c *DistroCriteria) Summarize() string {
 	return "does not match distro(s): " + strings.Join(distroStrs, ", ")
 }
 
+type ExactDistroCriteria struct {
+	Distros []distro.Distro
+}
+
+func (c *ExactDistroCriteria) MatchesVulnerability(value vulnerability.Vulnerability) (bool, string, error) {
```

Comment:
Oddly enough we probably don't want to depend on MatchesVulnerability() moving forward. The path for this is to really embed this within the v6 vulnerability provider (in the [query builder](https://github.com/anchore/grype/blob/main/grype/db/v6/search_query.go#L50-L84)). The reason for this is that there tends not to be enough information in the `vulnerability.Vulnerability` type to make these kinds of decisions without the v6 schema starting to creep through. In this case the code is leveraging namespaces, however, they are also on their way out also, and nothingn functionally depends on these values (they are here as a v5 hold over to keep the json presentation stable until grype v1).

Reply:
I'm not sure what is being suggested here. Can you elaborate a bit on what a better implementation would look like?

Reply:
ignore my comment it's just, all sorts of wrong